example/monteCarloIntegration/src/monteCarloIntegration.cpp

```diff
@@ -0,0 +1,207 @@
+ * SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR
+ * IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include <alpaka/alpaka.hpp>
+#include <alpaka/example/ExampleDefaultAcc.hpp>
+
+#include <iostream>
+#include <typeinfo>
+
+
+//#############################################################################
+//! This functor defines the function for which the integral is to be computed.
+struct Function
+{
+    //-----------------------------------------------------------------------------
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \param acc The accelerator to be executed on.
+    //! \param x The argument.
+    template<typename TAcc>
+    ALPAKA_FN_ACC
+    auto operator ()(
+        TAcc const & acc,
+        float const x) -> float
+    {
+        return alpaka::math::sqrt(
+            acc,
+            (1 - x * x));
+    }
+};
+
+//#############################################################################
+//! The kernel executing the parallel logic.
+//! Each Thread generates X pseudo random numbers and compares them with the given function.
+//! The local result will be added to a global result.
+struct Kernel
+{
+    //-----------------------------------------------------------------------------
+    //! The kernel entry point.
+    //! \tparam TAcc The accelerator environment to be executed on.
+    //! \tparam TFunctor A wrapper for a function.
+    //! \param acc The accelerator to be executed on.
+    //! \param numPoints The total number of points to be calculated.
+    //! \param globalCounter The sum of all local results.
+    //! \param functor The function for which the integral is to be computed.
+    template<
+        typename TAcc,
+        typename TFunctor>
+    ALPAKA_FN_ACC
+    auto operator ()(
+        TAcc const & acc,
+        size_t const numPoints,
+        uint32_t *const globalCounter,
+        TFunctor functor) const -> void
+    {
+        // Get the global linearized thread idx.
+        auto const globalThreadIdx = alpaka::idx::getIdx<
+            alpaka::Grid,
+            alpaka::Threads>(acc);
+        auto const globalThreadExtent = alpaka::workdiv::getWorkDiv<
+            alpaka::Grid,
+            alpaka::Threads>(acc);
+
+        auto const linearizedGlobalThreadIdx = alpaka::idx::mapIdx<1u>(
+            globalThreadIdx,
+            globalThreadExtent);
+        // Setup generator and distribution.
+        auto generator = alpaka::rand::generator::createDefault(
+            acc,
+            linearizedGlobalThreadIdx[0],
+            0);
+        // For simplicity the interval is fixed at [0,1].
+        auto dist(alpaka::rand::distribution::createUniformReal<float>(acc));
+
+        uint32_t localCount = 0;
+        for(size_t i = globalThreadIdx[0];
```

Comment:
I think this is a mistake and should be `linearizedGlobalThreadIdx[0]` (or without `[0]` with my earlier suggestion) instead.